uk.co.compucorp.civicrm.civihrsampledata/CRM/CiviHRSampleData/Importers/AbsencePeriod.php

```diff
@@ -1,31 +1,40 @@
 <?php
 
-
 /**
- * Class CRM_CiviHRSampleData_Importers_AbsencePeriod
+ * Class CRM_CiviHRSampleData_Importer_AbsencePeriod
  *
  */
-class CRM_CiviHRSampleData_Importers_AbsencePeriod extends CRM_CiviHRSampleData_DataImporter
+class CRM_CiviHRSampleData_Importer_AbsencePeriod extends CRM_CiviHRSampleData_DataImporter
 {
 
   public function __construct() {
     $this->removeAllPeriods();
   }
 
   /**
-   * @see CRM_CiviHRSampleData_DataImporter::insertRecord
+   * {@inheritdoc}
+   *
    * @param array $row
    */
   protected function insertRecord(array $row) {
     $this->callAPI('HRAbsencePeriod', 'create', $row);
   }
 
   /**
-   * Remove any existing absence period.
+   * Removes existing absence period.
    */
   private function removeAllPeriods() {
-    $query = "DELETE FROM civicrm_hrabsence_period";
-    CRM_Core_DAO::executeQuery($query);
+    $absencePeriods = $this->callAPI('HRAbsencePeriod', 'get', [
+      'sequential' => 1,
+      'return' => ["id"],
+      'options' => ['limit' => 0],
+    ]);
+
+    foreach($absencePeriods['values'] as $absencePeriod) {
+      $this->callAPI('HRAbsencePeriod', 'delete', [
+        'id' => $absencePeriod['id'],
+      ]);
+    }
```

Comment:
yup because there will not be a lot of periods ( don't think more than one ) , so  I decided it is better to stick with the API.